src/stores/ClientPresenceStore.js

```diff
@@ -0,0 +1,55 @@
+const PresenceStore = require('./DataStore');
+const Collection = require('../util/Collection');
+const Constants = require('../util/Constants');
+const { Presence } = require('../structures/Presence');
+
+class ClientPresenceStore extends PresenceStore {
+  constructor(...args) {
+    super(...args);
+    this.clientPresence = new Presence({
+      status: 'online',
+      afk: false,
+      since: null,
+      activity: null,
+    });
+  }
+
+  async setClientPresence({ status, since, afk, activity }) {
+    const applicationID = activity.application ? activity.application.id || activity.application : null;
+    let assets = new Collection();
+    if (activity.assets && applicationID) {
+      try {
+        const a = await this.client.api.applications(applicationID).assets.get();
```

Comment:
This should be `this.client.api.oauth2.applications(applicationID).assets.get()` (missing `oauth2`).